### DISCUSSION TITLE
[2023-10-09] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#8000] [Weekly Demo Day]
- James: Using conversions in a few places #7859
- James: Working on Enso Cloud Integration (Secrets and Connectors). #8006 
- James: Prototype of `devive_value` and `iif` on a `Table`. #8005 
- Radosław: Fix `NumberParser` to avoid `thousandSeparator==decimalPoint` and prefer US decimal format 
  #7946
- Radosław: Implement `Table.lookup_and_replace` for in-memory 
  #7979
- Radosław: [DRAFT] Refactoring Table problem handling
  #7879
  - Got another big piece of the refactor done, more than halfway through with it. The new approach looks neat IMHO.
- Radosław: Fix a bug that could cause loss of data in case of a race condition
  #8007
- Greg: Basic XML support #7947
- Greg: Cache filtered children in XML #7998
- Cassandra: Add Text.substring to allow for an easy short hand of Text.take (start.up_to end) #7913
- Cassandra: Change filter_blank_rows when_any parameter to have a more user-friendly type #7935
- Jaroslav: Instrumenter to observe behavior of nodes with UUID #7833
- Jaroslav: Meta.Type.find a type by FQN #7885
- Jaroslav: Avoid classloading deadlock by delaying computations of other passes #7943
- Jaroslav: Using ENSO_JAVA env variable to turn on the Espresso support #7951
- Jaroslav: Avoid unnecessary conversions in favor of direct match #7952
- Jaroslav: Removal of unused onExceptionalCallback #7970
- Hubert: Enable log-to-file configuration #7918
- Hubert: Add support for https and wss #7937
- Hubert: Remove context locks from visualization commands #7953
- Hubert: Native image should not be tied a particular hardware #7956
- Hubert: Fix initialization race-condition in runtime connector #7985
- Hubert: Investigating CI regression (merged as part of #7953)
- Pavel: Implement private modules #7840
- Pavel: Exporting non-existing symbols fails with compiler error #7960
- Pavel: Upgrade enso to GraalVM for jdk 21 #7991
- Dmitry: Open projects using the packaged language server #7868
- Dmitry: Fix bundled GraalVM path #7948
- Dmitry: Remove extra dot in error messages #7974
- Dmitry: Restore execution environment before the execution complete notification #7980

**Comment by radeusgd** [comment]
Lookup and Replace:
![image](https://github.com/enso-org/enso/assets/1436948/5075ca3d-5aa3-4cc1-a613-c43c6e29a2f5)
